### PR TITLE
feat(ST-95): add card component and display class materials

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added a resources container and added a dynamic arrow for open and closing materials display section
- Created a card component for displaying resources
- Added style file
- Added a get resources rtk query
- Integrated with the resources container component

## Associated Ticket Link(s)

- https://trello.com/c/LtoynaFM

## Related Pull Request(s)

- N/A

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/fe6dce83-e3c8-44ce-a938-0b45069ac279

### CODE REVIEW

**Review by ragib-hossain-sazim — APPROVED** [review]

File: modules/Classroom/components/ResourceMaterialCard/ResourceMaterialCard.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import { useResourceCardStyles } from "./ResourceMaterialCard.styles";
+import { TResourceMaterialCardType } from "./ResourceMaterialCard.types";
+
+const ResourceMaterialCard = ({ resource }: TResourceMaterialCardType) => {
+  const { classes } = useResourceCardStyles();
+
+  const handleOpenFile = () => {
+    window.open(resource.fileUrl, "_blank");
+  };
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/components/ResourceMaterialCard/ResourceMaterialCard.tsx:12]: As discussed in standup, add a ticket to integrate download file functionality. Meanwhile add  `// TODO` comment here and other places where file download relevant.

File: modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { useGetResourceMaterialsQuery } from "@/shared/redux/rtk-apis/resources/resources.api";
+
+import ResourceMaterialCard from "../../components/ResourceMaterialCard/ResourceMaterialCard";
+import { useResourcesStyles } from "./ClassroomResourcesContainer.styles";
+import { TClassroomResourcesContainerProps } from "./ClassroomResourcesContainer.types";
+
+const ClassroomResourcesContainer = ({ classroomId }: TClassroomResourcesContainerProps) => {
+  const { classes } = useResourcesStyles();
+  const [isMaterialsOpened, { toggle: materialsToggle }] = useDisclosure(false);
+  const { data: fetchedMaterials, isLoading: isMaterialsLoading } =
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx:15]: Query should be skipped when classroomId is falsy

File: modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    useGetResourceMaterialsQuery(classroomId);
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx:16]: Query should be skipped when classroomId is falsy

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left some comments

**Pull request merged**